### PR TITLE
Filtering update

### PR DESCRIPTION
**Pull request by Stokalas**
Since to keep up to date I merged dynamic_monthly branch into this it shows it's changes as well. Only file to look into is DataFilter.cs. Other files MUST be reviewed in pull request for dynamic_monthly.

**Added filtering options:**

- Income/expenses until today
- Income/expenses until end of this month
- Income/expenses from certain date until certain date
- Income/expenses from certain date until today

### CODE REVIEW

**Comment by Stokalas** [comment]
I'm not sure if filtering until end of this month has to include all entries before this month as well, or only this month and no earlier entries 🙂

File: SmartSaver/DataManagement/MonthlyUpdater.cs [hunk 1]
```diff
@@ -18,23 +18,23 @@ public MonthlyUpdater(DataFilter dataFilter, Data data)
         }
         public void UpdateMonthlyEntries(int userId) //eina per visus
         {
-            var entriesInc = _dataFilter.GetRecurringIncome();
-            var entriesExp = _dataFilter.GetRecurringExpenses();
+            var entriesInc = _dataFilter.GetIncome(isMonthly:true);
+            var entriesExp = _dataFilter.GetExpenses(isMonthly: true);
```
Comment (ciliakas) [inline SmartSaver/DataManagement/MonthlyUpdater.cs:22]: Nice touch with the named argument usage, I think this way with the renamed methods the filtering is now more clear and easier to use

**Review by ciliakas — APPROVED** [review]
Everything that's in here looks fine, plenty of filtering options to choose from, if we ever need more we can just add them as we go along later.

**Review by MatasLa — APPROVED** [review]
Totally agree with Arnas, this way of renaming methods seems more clear and easier to use. Also great job with adding more filtering options.

**Pull request merged**